### DISCUSSION TITLE
Docs: Testing: Integration example

### DISCUSSION THREAD

**Discussion by nimmolo** [rmosolgo/graphql-ruby#3908] [Q&A] [answered]
In the GraphQL-Ruby documentation examples, there are many examples referring to the `context: { current_user }`. In the [Integration Tests](https://graphql-ruby.org/testing/integration_tests.html) page, this is generally `context: { viewer }`. I understand that the `context` properties are arbitrary and/or conventional. However, in one case it's tripping me up a little. 

In the example for integration tests for [Transport-level behaviors](https://graphql-ruby.org/testing/integration_tests.html#testing-transport-level-behaviors):

```ruby
it "loads user token into the viewer" do
  query_string = "{ viewer { username } }"
```
Is `viewer` here assumed to be a Query type returning a `User` if there's a `current_user` as determined by the `graphql_controller`?  Or is the above query string looking for `viewer` in the `context` object (analogous to a `current_user` elsewhere in the docs examples)? 

I am thinking it's the former, because I believe the controller is the sole source of the context object, and that it is never returned in the JSON. (Can you even query the `context` object in a query string? I don't imagine so.)

**Comment by rmosolgo** [comment, marked as answer]
In that example, it's imagined that `viewer` is this: 

```ruby
class Types::Query 
  field :viewer, User
  
  def viewer 
    context[:current_user]
  end 
end 
```

That is, a field on the `Query` type that returns the currently-authenticated user, if there is one. This convention started with Relay.js, which, IIRC, _required_ such a field to be implemented. At least GitHub's API still has a field like that.

**Reply by nimmolo** [reply]
Thanks!
